stdlib/string/templatelib.pyi

```diff
@@ -0,0 +1,28 @@
+from collections.abc import Iterator
+from typing import Any, Literal, final
+
+__all__ = ["Interpolation", "Template"]
+
+@final
+class Template:
+    strings: tuple[str, ...]
+    interpolations: tuple[Interpolation, ...]
+
+    def __new__(cls, *args: str | Interpolation) -> Template: ...
+    def __iter__(self) -> Iterator[str | Interpolation]: ...
+    def __add__(self, other: Template | str) -> Template: ...
+    @property
+    def values(self) -> tuple[Any, ...]: ...
+
+@final
+class Interpolation:
+    value: Any  # TODO: consider making `Interpolation` generic in runtime
+    expression: str
+    conversion: Literal["a", "r", "s"] | None
```

Comment:
Mini rant: Why they used these arcane letters instead of just `"repr"`, `"str"`, and `"ascii"` is beyond me.